source/custom/custom.go

```diff
@@ -141,6 +148,16 @@ func (s Source) discoverFeature(feature FeatureSpec) (bool, error) {
 				continue
 			}
 		}
+		// hostname rule
```

Comment:
not related to this commit, but we could possibly do something nifty with reflect package to avoid this repetitive code.

Reply:
I gave it a try. It works, but the code is a bit ugly IMHO. But that might be personal preference. You can see it at [1].
In this PR I went with a different approach for code de-deduplication, see last commit. It still needs one line with reflection, I didn't find another to filter out the nil values, but overall its better readable. With the downside of still needing to touch it for new rules though.

WDYT?

[1] https://github.com/slintes/node-feature-discovery/commit/ad705f9e14d71508d367ba7cc83a13e0e198f0dd#diff-1b283f9d332ba27ed8b09eb6642c7bf9c9d552d45f89d485c590c7c1619f6b7dR101

Reply:
> I gave it a try. It works, but the code is a bit ugly IMHO

Doesnt seem too bad to me, but your current change is OK if you feel it is better.

Reply:
I think the refactor in the last patch of this PR is good. I'm not a big fan of reflect but I guess there's no nice and easy way around it here.

Reply:
I'll keep the current version version then :slightly_smiling_face: